worblehat-domain/src/main/java/de/codecentric/psd/worblehat/domain/BookService.java

```diff
@@ -11,7 +13,9 @@
 
 	void borrowBook(Book book, String borrower) throws BookAlreadyBorrowedException;
 
-	Book findBookByIsbn(String isbn);
+	void borrowOneBook(@Nonnull Set<Book> books, String borrower) throws BookAlreadyBorrowedException;

```

Comment:
I don't like this method. The problem is, that the implementation expects a set of already persisted books, but there is no way the call side can find out this without looking at the implementation. We should try to get rid of this after we merged this.

Reply:
I'll add a new method and mark the other one as deprecated. Also the controller will then only use the new method. It will look a bit messy though...